### PR TITLE
[PINOT-4726] Remove excessive connection-related logs from broker

### PR DESCRIPTION
**Pull request by mcvsubbu**

### CODE REVIEW

File: pinot-transport/src/main/java/com/linkedin/pinot/transport/scattergather/ScatterGatherImpl.java [hunk 1]
```diff
@@ -479,12 +485,12 @@ private void runInternal() {
           if (errorMap != null && errorMap.containsKey(_server)) {
             errStr = errorMap.get(_server).getMessage();
           }
-          LOGGER.warn("Destroying invalid conn {}:{}", conn, errStr);
+          LOGGER.info("Destroying invalid conn {}:{}", conn, errStr);
```
Comment (antumbde) [inline pinot-transport/src/main/java/com/linkedin/pinot/transport/scattergather/ScatterGatherImpl.java]: seems error. Are you taking approach of logging only non-recoverable issues as errors ?

File: pinot-transport/src/main/java/com/linkedin/pinot/transport/scattergather/ScatterGatherImpl.java [hunk 2]
```diff
@@ -501,14 +501,13 @@ private void runInternal() {
         LOGGER.debug("Response Future is : {}", _responseFuture);
         error = false;
       } catch (TimeoutException e1) {
-        LOGGER.info("Timed out waiting for connection for server ({})({})(gotConnection={}). Setting error future",
-            _server, _request.getRequestId(), gotConnection, e1.getMessage());
+        LOGGER.warn("Timed out waiting for connection for server ({})({})(gotConnection={}). See metric {}", _server,
+            _request.getRequestId(), gotConnection, e1.getMessage(), BrokerMeter.REQUEST_DROPPED_DUE_TO_CONNECTION_ERROR.toString());
```
Comment (antumbde) [inline pinot-transport/src/main/java/com/linkedin/pinot/transport/scattergather/ScatterGatherImpl.java]: this one has incorrect number of params. e1.getMessage() seems to be missed from log string

File: pinot-transport/src/main/java/com/linkedin/pinot/transport/scattergather/ScatterGatherImpl.java [hunk 3]
```diff
@@ -501,14 +501,13 @@ private void runInternal() {
...
         _responseFuture = new ResponseFuture(_server, e1, "Error Future for request " + _request.getRequestId());
-      } catch (RequestNotSentException e) {
-        LOGGER.info("Request {} not sent (gotConnection={}):{}. Setting error future",
-            _request.getRequestId(), gotConnection, e.getMessage());
+      } catch (ConnectionLimitReachedException e) {
+        LOGGER.warn("Request {} not sent (gotConnection={}):{}. See metric {}", _request.getRequestId(),
```
Comment (antumbde) [inline pinot-transport/src/main/java/com/linkedin/pinot/transport/scattergather/ScatterGatherImpl.java]: log server name

File: pinot-transport/src/main/java/com/linkedin/pinot/transport/scattergather/ScatterGatherImpl.java (hunk 1 above)
Comment (mcvsubbu) [inline pinot-transport/src/main/java/com/linkedin/pinot/transport/scattergather/ScatterGatherImpl.java]: Good catch, I will change that back to warn (if conn is not null).

**Pull request merged**